frontend/src/services/transactions/token.ts

```diff
@@ -3,7 +3,7 @@ import { BigNumber, Contract } from 'ethers';
 import { DeepReadonly } from 'vue';
 
 import StandardToken from '@/assets/StandardToken.json';
-import { EthereumProvider } from '@/services/web3-provider';
+import { useEthereumProvider } from '@/stores/ethereum-provider';
 
 export async function ensureTokenAllowance(
   signer: DeepReadonly<JsonRpcSigner>,
```

Comment:
Wouldn’t it be more consistent to also access the signer from the provider store?

Reply:
Good catch. I made also use of the store here. 👍🏾 

PS: We **really** need more strong type checks (I have a branch locally), because I forgot to adapt the function call too, which got not reported. 😑